bpf/include/bpf/ctx/xdp.h

```diff
@@ -33,28 +33,23 @@ static __always_inline __maybe_unused int
 xdp_load_bytes(const struct xdp_md *ctx, __u64 off, void *to, const __u64 len)
 {
 	void *from;
-	int ret;
-	/* LLVM tends to generate code that verifier doesn't understand,
-	 * so force it the way we want it in order to open up a range
-	 * on the reg.
-	 */
-	asm volatile("r1 = *(u32 *)(%[ctx] +0)\n\t"
-		     "r2 = *(u32 *)(%[ctx] +4)\n\t"
-		     "%[off] &= %[offmax]\n\t"
-		     "r1 += %[off]\n\t"
-		     "%[from] = r1\n\t"
-		     "r1 += %[len]\n\t"
-		     "if r1 > r2 goto +2\n\t"
-		     "%[ret] = 0\n\t"
-		     "goto +1\n\t"
-		     "%[ret] = %[errno]\n\t"
-		     : [ret]"=r"(ret), [from]"=r"(from)
-		     : [ctx]"r"(ctx), [off]"r"(off), [len]"ri"(len),
-		       [offmax]"i"(__CTX_OFF_MAX), [errno]"i"(-EINVAL)
-		     : "r1", "r2");
```

Comment:
Removing the inline asm here should likely result in verifier issues due to the way LLVM is generating the code. Originially, I had it similarly as you below, but had to switch to inline asm to force the pattern matching.

Reply:
@borkmann You are right, Thank you for this advice. I'm testing a new fix and will update that.